### PR TITLE
mds/Beacon: wake up the thread in shutdown()

### PR DESCRIPTION
**Pull request by MaxKellermann**
This eliminates the `wait_for()` delay and speeds up MDS shutdown.

This used to be part of https://github.com/ceph/ceph/pull/60320

## Checklist
- Tracker (select at least one)
  - [X] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [X] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [X] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [X] No tests

### CODE REVIEW

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
I'd also approve a ticket for backporting this.

**Comment by batrick** [comment]
jenkins test api

**Review by vshankar — APPROVED** [review]
@MaxKellermann please create a redmine ticket and link it in the commit message. Thx!

**Comment by MaxKellermann** [comment]
OK, I have registered an account and waiting for approval. Then I'll create tickets for several other PRs.

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68786](https://tracker.ceph.com/issues/68786).

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68859](https://tracker.ceph.com/issues/68859).

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/68859#note-2

**Pull request merged**
